File: app/src/main/java/com/cyl/musiclake/ui/music/playlist/PlaylistDetailContract.java
package com.cyl.musiclake.ui.music.playlist;

import android.content.Context;

import com.cyl.musiclake.base.BaseContract;
import com.cyl.musiclake.data.db.Music;
import com.cyl.musiclake.data.db.Playlist;

import java.util.List;

public interface PlaylistDetailContract {

    interface View extends BaseContract.BaseView {

        Context getContext();

        void showPlaylistSongs(List<Music> songList);

        void changePlayStatus(Boolean isPlaying);

        void removeMusic(int position);

        void success(int type);
    }

    interface Presenter extends BaseContract.BasePresenter<View> {

        void loadPlaylistSongs(Playlist playlist);

        void deletePlaylist(Playlist playlist);

        void renamePlaylist(Playlist playlist, String title);

        void disCollectMusic(String pid, int position, Music music);
    }
}

File: app/src/main/java/com/cyl/musiclake/ui/music/playlist/PlaylistDetailContract.kt
package com.cyl.musiclake.ui.music.playlist

import com.cyl.musiclake.bean.Album
import com.cyl.musiclake.bean.Artist
import com.cyl.musiclake.bean.Music
import com.cyl.musiclake.bean.Playlist
import com.cyl.musiclake.ui.base.BaseContract

interface PlaylistDetailContract {

    interface View : BaseContract.BaseView {

        fun showPlaylistSongs(songList: MutableList<Music>?)

        fun showTitle(title: String) {}

        fun showCover(cover: String) {}

        fun showDescInfo(title: String) {}

        fun removeMusic(position: Int) {}

        fun success(type: Int)

        fun showEmptyView(msg: String)
    }

    interface Presenter : BaseContract.BasePresenter<View> {
        fun loadPlaylistSongs(playlist: Playlist)

        fun loadArtistSongs(artist: Artist)

        fun loadAlbumSongs(album: Album)

        fun deletePlaylist(playlist: Playlist)

        fun renamePlaylist(playlist: Playlist, title: String)
    }
}
